dz5/home/views.py

```diff
@@ -1,26 +1,52 @@
 from django.shortcuts import render, redirect
 from django.http import HttpResponse
 # Create your views here.
-
+from django.urls import reverse
+from django.views import View
 
 from home.models import Student
 from home.forms import StudentForm
 
 
-def home(request):
-    if request.method == "GET":
-        student = Student()
-        student.name = "Lana"
-        student.save()
+class HomeView(View):
+    def get(self, request):
         students = Student.objects.all()
         student_form = StudentForm()
 
-        context = {"students": students,
-                   'form': student_form,
+        context = {
+            "students": students,
+            "form": student_form,
                    }
         return render(request, "index.html", context=context)
-    elif request.method == "POST":
+
+    def post(self, request):
         student_form = StudentForm(request.POST)
         if student_form.is_valid():
             student_form.save()
-        return redirect("/home")
+        return redirect(reverse("students_class"))
+
+
+class UpdateView(View):
+
+    def get_student(self, id):
+
+        return Student.objects.get(id=id)
```

Comment:
If client pass wrong id you will get error, better use `get_object_or_404` function for that